### PR TITLE
Fixed all_shotest_paths key error  #1433

### PR DESCRIPTION
**Pull request by alexnikleo**
I decided to raise NetworkxNoPath exception.

### CODE REVIEW

**Comment by chebee7i** [comment]
Won't the key error raise in the if/else clause immediately before your modification? It seems to me the code should do a quick:

```
if not (source in G and target in G):
    raise Something
```

or something similar at the very beginning. I'm unsure of whether a NoPath vs KeyError is appropriate here. KeyError makes more sense to me, but perhaps others feel more strongly about it then I do. Maybe we should have a new exception UnknownNode.

**Comment by pebenito** [comment]
I use shortest_path, all_shortest_paths, and all_simple_paths in my project. Shortest_path and all_simple_paths appear to raise NetworkXError if the source node doesn't exist and NetworkXNoPath if the target node doesn't exist. That inconsistency is probably a separate issue, but my suggestion is that both situations should raise Error (or UnknownNode, if you go down that road).

**Comment by alexnikleo** [comment]
Agreed. We should choose one consistent way to say: "There is no node." I think it will be a good solution to add exception NetworkxUnknownObject. And use it in all places with missing node.

**Comment by chebee7i** [comment]
@alexnikleo UnknownObject doesn't provide a contextual interpretation. We know that a specified node is not present in the graph. Do you think that `NetworkXInvalidNode` is more informative? If so, let's try to create that instead, and then, update most of the functions in this module. So in the case that u and v are both `None`, then we should immediately check that they are both in the graph. If not, we raise the exception saying which of `u` or `v` or both is not a valid node in the graph.

**Comment by chebee7i** [comment]
Also, you will need to update the api/release notes found in the `doc/source/reference`

**Comment by chebee7i** [comment]
@alexnikleo are you still interested in this PR. Let us know if not, so we can proceed.

**Comment by alexnikleo** [comment]
@chebee7i yes, I'll add ```NetworkXInvalidNode``` exception in some days.

**Comment by alexnikleo** [comment]
@chebee7i, watch this PR, please. Is it ok?

**Comment by SanketDG** [comment]
@alexnikleo You have to rebase with the master branch. We don't have an `api_2.0.rst`, now its `release_2.0.rst`

File: networkx/algorithms/shortest_paths/generic.py [hunk 1]
```diff
@@ -150,7 +150,7 @@ def shortest_path_length(G, source=None, target=None, weight=None):
     source : node, optional
         Starting node for path.
         If not specified, compute shortest path lengths using all nodes as
-        source nodes.
+        source nodes.i
```
Comment (SanketDG) [inline networkx/algorithms/shortest_paths/generic.py]: Why is a `i` here?

**Comment by hagberg** [comment]
This needs some review and decision on if/how to complete.

**Pull request merged**